developers/blobstream-rollups.md

```diff
@@ -0,0 +1,123 @@
+when using a sequence of spans,
+the inclusion proofs will be straight from the rollup data up to the [data root tuple root](https://github.com/celestiaorg/blobstream-contracts/blob/master/README.md#how-it-works).
+However, in the case of using the `height` and the `share commitment`,
+an extra step would need to be done when posting the header to the settlement contract.
+This step is proving that the provided commitment is part of the Celestia block (referenced by its height).
+Then, fraud proofs will need to prove the following:
+
+- share inclusion to the share commitment: meaning creating two merkle proofs:
+  - share merkle proof up to the [subtree root](https://celestiaorg.github.io/celestia-app/specs/data_square_layout.html#blob-share-commitment-rules)
+    corresponding to that share
+  - subtree root merkle proof to the [share commitment](https://celestiaorg.github.io/celestia-app/specs/data_square_layout.html#blob-share-commitment-rules)
+- share commitment inclusion to the data root tuple root: meaning four merkle proofs:
+  - subtree roots merkle proofs to the share commitment: to make sure the subtree roots are valid
+  - subtree roots merkle proofs up to the row roots: to prove that the subtree roots belong to a set of rows
+  - row roots proofs to the data root
+  - data root tuple proof to the data root tuple
+
+Once these are valid, the rollup contract can proceed to parse the share and verify the contested claim.
+
+Note: **Generating/verifying share commitment proofs is still not supported.
+It still needs tooling to generate the proofs on the node side,
+and verifying them on the Solidity side which will be built in the upcoming months.
+Thus, the [sequence of spans](#optimistic-rollups-that-uses-a-sequence-of-spans) approach is preferable at the moment.
+Also, the gas cost of verifying all the above proofs needs to be investigated.**
+
+#### Protobuf deserialization contract
+
+Another alternative when using the share commitment to reference the rollup data is to parse the PFB,
+i.e., the protobuf serialized PayForBlob transaction,
+inside the rollup settlement contract and verify that the parsed share commitment is the same.
+Then, proving the inclusion of the PFB to the data root tuple root.
+More details on this can be found in [ADR-011](https://github.com/celestiaorg/celestia-app/blob/main/docs/architecture/adr-011-optimistic-blob-size-independent-inclusion-proofs-and-pfb-fraud-proofs.md). 
+
+The issue with this approach is needing a library to parse the protobuf in solidity and the gas cost that comes with it.
+Otherwise, proving the PFB to the data root tuple root is cheaper than proving the blob,
+because it needs only proving two shares.
+
+## Zk-Rollups
+
+Zk-rollups, aka validity rollups, can also use Celestia as a DA and Blobstream to verify that the data was posted.
+However, the submission process is different from the above constructions,
```

Comment:
after listing/describing the different ways to prove data, then I think that we can use a sentence similar to 

> However, the submission process is different from the above constructions

something along these lines, where can start the rollup section with "How data inclusion is proved is highly dependant upon the rollup's proving system. Then discuss ORU, then ZKR